GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -320,6 +326,127 @@ public void run() {
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayArray[index] = Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length);
+    }
+
+    private void renderSixCamerasAndReadback(final GVRCameraRig mainCameraRig,
+            ByteBuffer readbackBuffer, byte[][] byteArrayArray) {
+        if (byteArrayArray.length != 6) {
+            throw new IllegalArgumentException(
+                    "byteArrayArray length is not 6.");
+        } else {
+            // temporarily create a center camera
+            GVRPerspectiveCamera centerCamera = new GVRPerspectiveCamera(this);
+            centerCamera.setFovY(90.0f);
+            centerCamera.setRenderMask(GVRRenderMaskBit.Left
+                    | GVRRenderMaskBit.Right);
+            GVRSceneObject centerCameraObject = new GVRSceneObject(this);
+            centerCameraObject.attachCamera(centerCamera);
+            mainCameraRig.getOwnerObject().addChildObject(centerCameraObject);
+            GVRTransform centerCameraTransform = centerCameraObject
+                    .getTransform();
+
+            int index = 0;
+            // render +x face (the one user is currently looking at)
+            centerCameraTransform.rotateByAxis(-90, 0, 1, 0);
+            renderOneCameraAndAddToList(centerCamera, readbackBuffer,
+                    byteArrayArray, index++);
+
+            // render -x face
+            centerCameraTransform.rotateByAxis(180, 0, 1, 0);
+            renderOneCameraAndAddToList(centerCamera, readbackBuffer,
+                    byteArrayArray, index++);
+
+            // render +y face
+            centerCameraTransform.rotateByAxis(-90, 0, 1, 0);
+            centerCameraTransform.rotateByAxis(90, 1, 0, 0);
+            renderOneCameraAndAddToList(centerCamera, readbackBuffer,
+                    byteArrayArray, index++);
+
+            // render -y face
+            centerCameraTransform.rotateByAxis(180, 1, 0, 0);
+            renderOneCameraAndAddToList(centerCamera, readbackBuffer,
+                    byteArrayArray, index++);
+
+            // render +z face
+            centerCameraTransform.rotateByAxis(90, 1, 0, 0);
+            centerCameraTransform.rotateByAxis(180, 0, 1, 0);
+            renderOneCameraAndAddToList(centerCamera, readbackBuffer,
+                    byteArrayArray, index++);
+
+            // render -z face
+            centerCameraTransform.rotateByAxis(180, 0, 1, 0);
+            renderOneCameraAndAddToList(centerCamera, readbackBuffer,
+                    byteArrayArray, index++);
+
+            centerCameraObject.detachCamera();
+            mainCameraRig.getOwnerObject()
+                    .removeChildObject(centerCameraObject);
+        }
+    }
+
+    private void returnScreenshot3DToCaller(final GVRScreenshot3DCallback callback,
+            final byte[][] byteArrayArray, final int width, final int height) {
+
```

Comment:
Just a minor issue. The naming of byteArrayArray seems a little bit weird to me...Can it be any other name which may be a little bit more readable or meaningful?